Add EVM integration to docs (#202)
timewave-computer/valence-protocol: docs/src/authorizations_processors/authorization_user_actions.md

# User Actions

> - `send_msgs(label, vec[ProcessorMessage])`: users can run an authorization with a specific label. If the authorization is `Permissioned`, the authorization contract will check if they are allowed to execute it by checking that the user has the token in its wallet if it's `Permissioned (without limit)` or that the user sent the token along with the messages if it's `Permissioned (with limit)`. Along with the authorization label, the user will provide an array of encoded messages, together with the message type (e.g. `CosmwasmExecuteMsg`, `EvmCall`...) and any other parameters for that specific ProcessorMessage (e.g. for a `CosmwasmMigrateMsg` we need to also pass a code_id). The contract will then check that the messages match the ones defined in the authorization (and in the correct order) and that all Parameters restrictions, if applied, are correct.

Reviewer: ```suggestion
- `send_msgs(label, vec[ProcessorMessage])`: users can run an authorization with a specific label. If the authorization is `Permissioned`, the authorization contract will check if their account is allowed to execute by checking that the account holds the token in its wallet. If the authorization is `Permissioned (without limit)` execution can be carried out indefinitely. If the authorization is is `Permissioned (with limit)` they may attempt execution any number of times up to that limit. Along with the authorization label, the user will provide an array of encoded messages, together with the message type (e.g. `CosmwasmExecuteMsg`, `EvmCall`, etc.) and any other parameters for that specific ProcessorMessage (e.g. for a `CosmwasmMigrateMsg` we need to also pass a code_id). The contract will then check that the messages match those defined in the authorization, that the messages appear in correct order, and that any applied parameter restrictions are correct.
```

Author: I rephrased this in a different way as the correction was not right. If the authorization is permissioned (with limit), the account must attach the authorization token along with the messages. If its permissioned (without limit) we just query the account to see if it has the token but they don't have to send it. Let me know if it's OK

> - `send_msgs(label, vec[ProcessorMessage])`: users can run an authorization with a specific label. If the authorization is `Permissioned (without limit)`, the authorization contract will check if their account is allowed to execute by checking that the account holds the token in its wallet, and it can be executed indefinitely. If the authorization is is `Permissioned (with limit)` the account must attach the authorization token to the contract execution. Along with the authorization label, the user will provide an array of encoded messages, together with the message type (e.g. `CosmwasmExecuteMsg`, `EvmCall`, etc.) and any other parameters for that specific ProcessorMessage (e.g. for a `CosmwasmMigrateMsg` we need to also pass a code_id). The contract will then check that the messages match those defined in the authorization, that the messages appear in correct order, and that any applied parameter restrictions are correct.

Reviewer 2: ```suggestion
- `send_msgs(label, vec[ProcessorMessage])`: users can run an authorization with a specific label. If the authorization is `Permissioned (without limit)`, the authorization contract will check if their account is allowed to execute by checking that the account holds the token in its wallet, and it can be executed indefinitely. If the authorization is `Permissioned (with limit)` the account must attach the authorization token to the contract execution. Along with the authorization label, the user will provide an array of encoded messages, together with the message type (e.g. `CosmwasmExecuteMsg`, `EvmCall`, etc.) and any other parameters for that specific ProcessorMessage (e.g. for a `CosmwasmMigrateMsg` we need to also pass a code_id). The contract will then check that the messages match those defined in the authorization, that the messages appear in correct order, and that any applied parameter restrictions are correct.
```

> - `send_msgs(label, vec[ProcessorMessage])`: users can run an authorization with a specific label. If the authorization is `Permissioned (without limit)`, the authorization contract will check if their account is allowed to execute by checking that the account holds the token in its wallet, and it can be executed indefinitely. If the authorization is is `Permissioned (with limit)` the account must attach the authorization token to the contract execution. Along with the authorization label, the user will provide an array of encoded messages, together with the message type (e.g. `CosmwasmExecuteMsg`, `EvmCall`, etc.) and any other parameters for that specific ProcessorMessage (e.g. for a `CosmwasmMigrateMsg` we need to also pass a code_id). The contract will then check that the messages match those defined in the authorization, that the messages appear in correct order, and that any applied parameter restrictions are correct.

Reviewer 2: What does `and it can be executed indefinitely` refer to here? Are you just saying that the subroutine is authorized to execute without limit? If that's the case, then this redundant as it's already predicated in the sentence.

Author: Right, removed it.

Author: right, removed it

  If all checks are correct, the contract will route the messages to the correct `Processor` with an `execution_id` for the processor to callback with. This `execution_id` is unique for the entire application.

>   If the execution of all the actions are confirmed via a callback, we will burn the token and if they fail, we will send the token back.

Reviewer: ```suggestion
  If execution of all actions is confirmed via a callback the authorization token is burned. If execution fails, the token is sent back.
```

>   If execution of all actions is confirmed via a callback the authorization token is burned. If execution fails, the token is sent back.

Reviewer 2: ```suggestion
  If execution of all actions is confirmed via a callback, the authorization token is burned. If execution fails, the token is sent back.
```
I think you need this comma.
